sklearn/utils/validation.py

```diff
@@ -429,6 +432,10 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         msg = ("Data with input dtype %s was converted to %s%s."
                % (dtype_orig, array.dtype, context))
         warnings.warn(msg, DataConversionWarning_)
+
+    if copy and array is array_orig:
```

Comment:
Also I think that might fix the transformer part for the https://github.com/scikit-learn/scikit-learn/issues/5481